internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalNameHashing.scala

```diff
@@ -32,8 +32,10 @@ private final class IncrementalNameHashing(log: sbt.util.Logger, options: IncOpt
       val aNameHashes = a.nameHashes
       val bNameHashes = b.nameHashes
       val modifiedNames = ModifiedNames.compareTwoNameHashes(aNameHashes, bNameHashes)
-      val apiChange = NamesChange(className, modifiedNames)
-      Some(apiChange)
+      if (modifiedNames.regularNames.nonEmpty || modifiedNames.implicitNames.nonEmpty) {
```

Comment:
I've encountered situations where for some reason class got different API with no names changed (on 'normal' incremental compilation - not 'cached' ones).

Debugging problems with API hashes is really hard so I gave up looking why class has changed API without any names change and just introduce this check.